src/websites/controllers/domains.controller.ts

```diff
@@ -55,44 +56,50 @@ export class DomainsController {
       throw new BadRequestException('No website found for this campaign')
     }
 
-    return this.domains.startDomainRegistration(user, website.id, domain)
-  }
+    const domain = await this.domains.getDomainWithPayment(website.id)
 
-  @Post('complete')
-  @UseCampaign()
-  @HttpCode(HttpStatus.OK)
-  async completeDomainRegistration(
-    @ReqCampaign() { id: campaignId }: Campaign,
-  ) {
-    const website = await this.websites.findUnique({
-      where: { campaignId },
-      select: { id: true },
-    })
-
-    if (!website) {
-      throw new BadRequestException('No website found for this campaign')
+    if (!domain) {
+      return {
+        message: DomainOperationStatus.NO_DOMAIN,
+        paymentStatus: null,
+      }
     }
 
-    // TODO: remove and use body https://goodparty.atlassian.net/browse/WEB-4233
-    const dummyContact: RegisterDomainSchema = {
-      firstName: 'John',
-      lastName: 'Doe',
-      email: 'test@example.com',
-      phoneNumber: '1234567890',
-      addressLine1: '123 Main St',
-      addressLine2: 'Apt 1',
-      city: 'Anytown',
-      state: 'CA',
-      zipCode: '12345',
+    let paymentStatus: PaymentStatus | null = null
+    if (domain.paymentId) {
+      paymentStatus = await this.domains.getPaymentStatus(domain.paymentId)
     }
 
-    return this.domains.completeDomainRegistration(website.id, dummyContact)
-  }
+    // With Vercel, we can directly use the domain status from our database
+    // since Vercel operations are processed immediately
+    let message: DomainOperationStatus
+    switch (domain.status) {
+      case DomainStatus.pending:
+        message = DomainOperationStatus.SUBMITTED
+        break
+      case DomainStatus.submitted:
+        message = DomainOperationStatus.IN_PROGRESS
+        break
+      case DomainStatus.registered:
+        message = DomainOperationStatus.SUCCESSFUL
+        break
+      case DomainStatus.inactive:
+        message = DomainOperationStatus.ERROR
+        break
+      default:
+        message = DomainOperationStatus.ERROR
+    }
 
-  @Get('status')
-  @UseCampaign()
-  async checkRegistrationStatus(@ReqCampaign() { id: campaignId }: Campaign) {
-    return this.domains.checkRegistrationStatus(campaignId)
+    return {
+      message,
+      paymentStatus,
+      operationDetail: {
+        operationId: domain.operationId,
+        status: message,
+        type: 'RegisterDomain',
```

Comment:
Should this be in an `enum` somewhere?

Not sure what this is used for since it's a static value that doesn't appear to change for this given endpoint 🤔